contracts/Recoverable.sol

```diff
@@ -1,5 +1,133 @@
 pragma solidity ^0.4.24;
 
-interface Recoverable {
 
+contract Recoverable {
+  address public owner_;
+  address private recoverer_;
+  uint256 private recoveryTimeRequred_;
+  uint256 private timeOfRecovery_;
+
+  event OwnerChanged(
+    address indexed previousOwner,
+    address indexed newOwner
+  );
+  event RecovererChanged(
+    address indexed owner_,
+    address indexed newRecoverer
+  );
+  event RecoveryStarted(
+    address indexed owner_,
+    address indexed heir,
+    uint256 timeOfRecovery
+  );
+  event Recovered(
+    address indexed previousOwner,
+    address indexed newOwner
+  );
+  event OwnerBlockedRecovery(address indexed owner_);
+
+  /**
+   * Check that a function is called by the owner_
+   */
+  modifier isOwner() {
+    require(msg.sender == owner_);
+    _;
+  }
+
+  /**
+   * Check that a function is called by the recoverer_
+   */
+  modifier isRecoverer() {
+    require(msg.sender == recoverer_);
+    _;
+  }
+
+  /**
+   * Creates a recoverable contract with owner set to the original sender and
+   * required recovery time set by the original sender
+   */
+  constructor(uint256 _recoveryTime) public {
+    owner_ = msg.sender;
+    recoveryTimeRequred_ = _recoveryTime;
+  }
+
+  /**
+   * The owner can set a new recoverer when they wish
+   */
+  function setRecoverer(address _newRecoverer) public isOwner {
+    require(_newRecoverer != owner_);
+    emit RecovererChanged(owner_, _newRecoverer);
+    recoverer_ = _newRecoverer;
+  }
+
+  function recoverer() public view returns(address) {
+    return recoverer_;
+  }
+
+  function recoveryTimeRequred() public view returns(uint256) {
+    return recoveryTimeRequred_;
+  }
+
+  function timeOfRecovery() public view returns(uint256) {
+    return timeOfRecovery_;
+  }
+
+  /**
+   * Owner can remove the recoverer
+   */
+  function removeRecoverer() public isOwner {
+    recoverer_ = address(0);
+  }
+
+  /**
+   * Recoverer can start a recovery by announcing it.
+   * Recoverer will have to wait for the recovery time required.
+   */
+  function announceRecovery() public isOwner {
+    require(recoveryTimeLapsed());
+    emit RecoveryStarted(owner_, recoverer_, timeOfRecovery_);
+    timeOfRecovery_ = block.timestamp;
+  }
+
+  /**
+   * Owner can block recovery if they deem it illegitimate
+   */
+  function blockRecovery() public isOwner {
+    emit OwnerBlockedRecovery(owner_);
+    timeOfRecovery_ = 0;
+  }
+
+  /**
+   * Once reverty time has lapsed, a recoverer may transfer ownership
+   * to themselves
+   */
+  function recoverContract() public isRecoverer {
```

Comment:
1. Whether the recoverer is set to owner or the funds are sent to their address, they are at the end of the day the owner until recovery by the original owner. Seems to me it wouldn't matter, and this is the truest implementation of a "recoverable" wallet. If the funds are sent somewhere else, the wallet isn't really recovered, it's abandoned.
2. I don't understand this one. We are creating this Recoverable wallet for someone to own. I suppose it could be extended for multiple owners in the future, but we don't need to do that here.
3. This doesn't change anything. The recoverer could then send the funds somewhere if the rest of the wallet is broken. We should probably make it unbreakable tho...